src/python/zquantum/qaoa/estimators.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+from openfermion import IsingOperator
+from typing import Optional
+from overrides import overrides
+
+from zquantum.core.circuit import Circuit
+from zquantum.core.interfaces.backend import QuantumBackend, QuantumSimulator
+from zquantum.core.measurement import ExpectationValues, Measurements
+from zquantum.core.interfaces.estimator import Estimator
+
+
+class CvarEstimator(Estimator):
+    """An estimator for calculating expectation value using CVaR method.
+    The main idea is that for diagonal operators the ground state of the Hamiltonian is a base state.
+    In particular for the combinatorial optimization problems, we care only about getting a single bitstring representing the solution.
+    Therefore, it might be beneficial for the optimization process to discard samples that represent inferior solutions.
+    CVaR Estimator takes uses only a top X percentile of the samples for calculating the expectation value, where X = alpha*100.
+
+    Reference: https://arxiv.org/abs/1907.04769
+    "Improving Variational Quantum Optimization using CVaR", P. Barkoutsos, G. Nannicini, A. Robert, I. Tavernelli, and S. Woerner
+    """
+
+    def __init__(self, alpha):
+        super().__init__()
+        self.alpha = alpha
+
+    @overrides
+    def get_estimated_expectation_values(
+        self,
+        backend: QuantumBackend,
+        circuit: Circuit,
+        target_operator: IsingOperator,
+        n_samples: Optional[int] = None,
+    ) -> ExpectationValues:
+        """Given a circuit, backend, and target operators, this method produces expectation values
+        using CVaR algorithm.
+
+        Args:
+            backend (QuantumBackend): the backend that will be used to run the circuit
+            circuit (Circuit): the circuit that prepares the state.
+            target_operator (SymbolicOperator): Operator to be estimated.
+            alpha (float): defines what part of the best measurements should be taken into account in the estimation process.
+            n_samples (int): Number of measurements done on the unknown quantum state.
+
+        Raises:
+            AttributeError: If backend is not a QuantumSimulator.
+
+        Returns:
+            ExpectationValues: expectation values for each term in the target operator.
+        """
+        if self.alpha > 1 or self.alpha <= 0:
+            raise ValueError("alpha needs to be a value between 0 and 1.")
+
+        if type(target_operator) is not IsingOperator:
```

Comment:
As usually, I recommend to use `isinstance(target_operator, IsingOperator)` instead of direct type comparison.

Reply:
Fixed!